server/homebrew.api.js

```diff
@@ -257,17 +257,22 @@ const deleteGoogleBrew = async (account, id, editId, res)=>{
 };
 
 const deleteBrew = async (req, res)=>{
-	const { brew, account } = req;
-	if(brew.googleId) {
-		const googleDeleted = await deleteGoogleBrew(account, brew.googleId, brew.editId, res)
+	let brew = req.brew;
+	const { googleId, editId } = brew;
+	const account = req.account;
+	const isOwner = account && (brew.authors.length === 0 || brew.authors[0] === account.username);
+	console.log(brew.authors, googleId, account.username);
+	let afterSave = async ()=>true;
+	if(googleId && isOwner) {
+		afterSave = async ()=>await deleteGoogleBrew(account, googleId, editId, res)
```

Comment:
Can we add a comment here? Just to be consistent with the comments below that help outline each of the cases we have to handle.

Reply:
Hmmm.... Trying to work through a clearer branching flow here. While the `afterSave()` approach works, it does scramble the order around a bit to be a little harder to read. What if we simply set a flag `let shouldDeleteFromGoogle = true` or something? Then I think we can rearrange the logic a bit for legibility:

```
If not on HB (Google Brew only):

  -> Flag to delete from Google

Else (must be on HB as a full brew or stub):

  If current user is only author:

    -> Delete from HB
    -> Flag to delete from Google (if stub)

  Else (multiple authors):

    If current user is owner of Google Brew:

      -> Update stub to full brew
      -> Flag to delete from Google

    -> Remove current user as author
    -> Save brew to HB with updated author list

    After save successful:

      If flag to delete from Google:

        -> Delete from Google
```